pgtoolkit/log/syslog.py

```diff
@@ -44,24 +51,42 @@ def build_prefix_re(self, syslog_sequence_numbers):
             pattern += r"\[(?P<chunk_nr>\d+)\] "
         return pattern
 
-    def parse(self, lines):
+    def process(self, lines):
+        """Preprocess syslog lines for parsing
+
+        If a line matches syslog prefix, the preprocessor wraps it unprefixed
+        with :class:`SyslogLine` object. This object hold decoded line as
+        PostgreSQL would have written it to a log file as well as syslog
+        metadata (timestamp, pid, etc.)
+
+        :param lines: A line iterator such as a file object.
+        :returns: Yields each :class:`SyslogLine` or plain line.
+
+        """
         for line in lines:
             m = self._re.match(line)
             if m:
                 message = line[len(m.group(0)):]
-                yield SyslogLine(
-                    message, date_parser=self.date_parser, **m.groupdict())
-            else:
-                yield line
+                if message.startswith('#011'):
+                    # Replace leading #011 by native tabulation.
+                    message = '\t' + message[4:]
+                line = SyslogLine.factory(
+                    message,
+                    date_parser=self.date_parser,
+                    **m.groupdict()
+                )
+            yield line
 
 
-class SyslogLine(UserString):
-    def __init__(self, message, date_parser=None, **kw):
-        if message.startswith('#011'):
-            message = '\t' + message[4:]
+class SyslogLine(str):
+    @classmethod
+    def factory(cls, message, date_parser=None, **kw):
+        # str.__init__ does not accepts more than one parameter. Let's use a
+        # factory.
```

Comment:
I think this could be implemented by a `__new__` method. This which would also keep the possibility to use the class as `SyslogLine(m, date_parser=<...>, ...)`.